src/Database/Expression/WindowExpression.php

```diff
@@ -174,4 +271,27 @@ public function traverse(Closure $visitor)
 
         return $this;
     }
+
+    /**
+     * Builds frame offset sql.
+     *
+     * @param \Cake\Database\ValueBinder $generator Value binder
+     * @param int|null $offset Frame offset
+     * @param string $direction Frame offset direction
+     * @return string
+     */
+    protected function buildOffsetSql(ValueBinder $generator, ?int $offset, string $direction): string
+    {
+        if ($offset === 0) {
+            return 'CURRENT ROW';
+        }
+
+        $sql = sprintf(
+            '%s %s',
+            $offset ?? 'UNBOUNDED',
+            $direction
```

Comment:
FYI, the one issue with using direct string constants/types is allowing user-specified SQL to end up here. Both direction and type strings are used to build the sql string.

Users can force this to happen elsewhere so I don't know if that's an issue.